File: Android/app/src/main/java/com/dreamproject/inwords/util/ErrorBodyFormatter.java
package com.dreamproject.inwords.util;

import java.io.IOException;

import okhttp3.ResponseBody;
import retrofit2.HttpException;

public class ErrorBodyFormatter {
    public static String getErrorMessage(HttpException e){
        String message = "Undefined error";

        ResponseBody responseBody = e.response().errorBody();
        if (responseBody != null) {
            try {
                message = responseBody.string();
            } catch (IOException e1) {
                e1.printStackTrace();
            }
        }

        return message;
    }
}

File: Android/app/src/main/java/ru/inwords/inwords/data/ErrorBodyFormatter.kt
package ru.inwords.inwords.data

import retrofit2.HttpException
import java.io.IOException

fun getErrorMessage(e: HttpException): String =
        try {
            e.response()?.errorBody()?.string()
        } catch (e1: IOException) {
            e1.printStackTrace()
            null
        } ?: "Undefined error"


